barretenberg/cpp/src/barretenberg/stdlib/primitives/field/field_conversion.test.cpp

```diff
@@ -9,23 +10,30 @@ template <typename Builder> using fq = bigfield<Builder, bb::Bn254FqParams>;
 template <typename Builder> using bn254_element = element<Builder, fq<Builder>, fr<Builder>, curve::BN254::Group>;
 template <typename Builder> using grumpkin_element = cycle_group<Builder>;
 
-template <typename Builder> class StdlibFieldConversionTests : public ::testing::Test {
+template <typename Builder> class stdlib_field_conversion : public ::testing::Test {
   public:
-    template <typename T> void check_conversion(Builder& builder, T x)
+    // Serialize and deserialize
+    template <typename T> void check_conversion(T x, bool valid_circuit = true, bool point_at_infinity = false)
     {
         size_t len = bb::stdlib::field_conversion::calc_num_bn254_frs<Builder, T>();
         auto frs = bb::stdlib::field_conversion::convert_to_bn254_frs<Builder, T>(x);
         EXPECT_EQ(len, frs.size());
-        auto y = bb::stdlib::field_conversion::convert_from_bn254_frs<Builder, T>(builder, frs);
-        EXPECT_EQ(x.get_value(), y.get_value());
+        auto y = bb::stdlib::field_conversion::convert_from_bn254_frs<Builder, T>(frs);
+        bool expected = std::is_same_v<Builder, UltraCircuitBuilder> ? !point_at_infinity : true;
+
+        EXPECT_EQ(x.get_value() == y.get_value(), expected);
```

Comment:
nano-nit: in/out seems clearer than x/y in a context where x/y can be plausibly confused with point coordinates